### ISSUE TITLE
kdenlive: can't find frei0r-plugins (and other missing dependencies)

### ISSUE THREAD

**Issue by lambdafu** [NixOS/nixpkgs#29614]
## Issue description

kdenlive can't find the frei0r-plugins for effects.  The effects menu is empty.  I am using kdenlive without KDE using the following package list:

```
    kdenlive kinit frei0r breeze-icons dvdauthor vlc cdrkit 
```

You can also check under Settings->Run Config Wizard.

The kinit package is needed to be able to add video clips (otherwise you get an error about klauncher). The config wizard complains about the missing breeze icons, dvdauthor, vlc and cdrkit, but at least installing them seems to work.  For frei0r, I could not find a workaround (even munging LD_LIBRARY_PATH didn't do anything for me).

## Technical details

* System: 18.03.git.799435b (Impala)
* Nix version: nix-env (Nix) 1.11.14
* Nixpkgs version: 18.03.git.799435b7ca
* Sandboxing enabled: build-use-sandbox = false

**Comment by lambdafu** [comment]
With strace I could find the list of searched directories:

```
$ strace kdenlive 2>&1 | grep frei0r
open("/nix/store/vp0ndwc33bj1742pdly2d58kpxr2gvzy-mlt-6.4.1/lib/mlt/libmltfrei0r.so", O_RDONLY|O_CLOEXEC) = 11
open("/nix/store/vp0ndwc33bj1742pdly2d58kpxr2gvzy-mlt-6.4.1/share/mlt/frei0r/blacklist.txt", O_RDONLY) = 11
open("/nix/store/vp0ndwc33bj1742pdly2d58kpxr2gvzy-mlt-6.4.1/share/mlt/frei0r/param_name_map.yaml", O_RDONLY) = 11
read(11, "# MLT frei0r param name mapping "..., 4096) = 248
open("/home/marcus/.frei0r-1/lib", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/usr/local/lib/frei0r-1", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/opt/local/lib/frei0r-1", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/usr/lib64/frei0r-1", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/usr/lib/frei0r-1", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
```

Which allowed me this work around:

```
$ ls -l /home/marcus/.frei0r-1
lrwxrwxrwx 1 marcus users 77 Sep 21 16:48 /home/marcus/.frei0r-1 -> /nix/store/m6jn92f5vx9vjai0xc13gl2dinl69bbm-frei0r-plugins-1.6.1/lib/frei0r-1
```

Not great, but at least I can use kdenlive a bit...

**Comment by davidak** [comment]
![screenshot_2017-12-20_12-25-19](https://user-images.githubusercontent.com/91113/34205895-877d46a2-e584-11e7-8bd1-70013c0d7847.png)

```
[davidak@dc7800:~]$ strace kdenlive 2>&1 | grep "No such file"
access("/etc/ld-nix.so.preload", R_OK)  = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver/lib/tls/x86_64/x86_64/libKF5KIOFileWidgets.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/run/opengl-driver/lib/tls/x86_64/x86_64", 0x7ffd47751c70) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver/lib/tls/x86_64/libKF5KIOFileWidgets.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/run/opengl-driver/lib/tls/x86_64", 0x7ffd47751c70) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver/lib/tls/x86_64/libKF5KIOFileWidgets.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/run/opengl-driver/lib/tls/x86_64", 0x7ffd47751c70) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/run/opengl-driver/lib/tls/libKF5KIOFileWidgets.so.5", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/run/opengl-driver/lib/tls", 0x7ffd47751c70) = -1 ENOENT (No such file or directory)
```
More: https://gist.github.com/davidak/8718f84e245c7f6cd5ab6922abc1d3f8

---

 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.70, NixOS, 18.03pre123130.1bc288591ea (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-18.03pre123130.1bc288591ea"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by cx405** [comment]
Confirming this for "frei0r".


@davidak 
Can't confirm for "breeze" icons, I am on kde5 release, amd64, install 17.09 with upgrade to 18.03 channel.

FFmpeg can be installed via "ffmpeg-full".

**Comment by cx405** [comment]
@davidak  Thanks for strace!  
```
open("/home/davidak/.frei0r-1/lib", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/usr/local/lib/frei0r-1", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/opt/local/lib/frei0r-1", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/usr/lib64/frei0r-1", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/usr/lib/frei0r-1", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
```

this is result of "# updatedb && locate frei0r | less -X"
```
/nix/store/rdq9kykkpf0c0pimjf2rqcqvm7jzvv3j-system-path/lib/frei0r-1
/nix/store/rdq9kykkpf0c0pimjf2rqcqvm7jzvv3j-system-path/lib/gstreamer-1.0/libgstfrei0r.la
/nix/store/rdq9kykkpf0c0pimjf2rqcqvm7jzvv3j-system-path/lib/gstreamer-1.0/libgstfrei0r.so
/nix/store/rdq9kykkpf0c0pimjf2rqcqvm7jzvv3j-system-path/lib/pkgconfig/frei0r.pc
/nix/store/rdq9kykkpf0c0pimjf2rqcqvm7jzvv3j-system-path/share/doc/frei0r-plugins
/nix/store/z15112vx42jxgzl4c8k0kasdpri01a3d-system-path/lib/frei0r-1
/nix/store/z15112vx42jxgzl4c8k0kasdpri01a3d-system-path/lib/gstreamer-1.0/libgstfrei0r.la
/nix/store/z15112vx42jxgzl4c8k0kasdpri01a3d-system-path/lib/gstreamer-1.0/libgstfrei0r.so
/nix/store/z15112vx42jxgzl4c8k0kasdpri01a3d-system-path/lib/pkgconfig/frei0r.pc
/nix/store/z15112vx42jxgzl4c8k0kasdpri01a3d-system-path/share/doc/frei0r-plugins
```

apparently kdenlive searches for "frei0r" inside "/usr/lib", however the library is installed into "/lib".
needs file patching.

Possible workaround could be (untested!):
(pkgs.runCommand "ln-frei0r" { } ''
ln -s {/run/current-system/sw/lib/frei0r-1,$out/lib/frei0r-1} $out/usr/lib/frei0r-1
'')
right inside environment.systemPackages

in the spirit of: https://github.com/NixOS/nixpkgs/issues/27050#issuecomment-373171024
I am still learning my way with nix, sorry.

**Comment by RobbiNespu** [comment]
same with me
![2018-11-15_23-17](https://user-images.githubusercontent.com/3077875/48562209-b340c780-e92c-11e8-99d7-37454e9ff45b.png)

Fedora 29 KDE spins

**Comment by davidak** [comment]
I don't have dependency errors anymore. I'm now using GNOME 3.

![screenshot from 2019-02-13 07-15-47](https://user-images.githubusercontent.com/91113/52691088-06dca480-2f60-11e9-9a77-4455655698a9.png)

---

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.98, NixOS, 18.09.2137.168cbb39691 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(davidak): `""`
 - channels(root): `"nixos-18.09.2137.168cbb39691, nixos-hardware, nixos-unstable-19.03pre169108.36f31600749"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by jtojnar** [comment]
Won't this work?

https://github.com/NixOS/nixpkgs/blob/6dea8fe56e0341b57f441be9d1dcfb59cfe59b1c/pkgs/applications/video/shotcut/default.nix#L41

**Comment by cx405** [comment]
@jtojnar  I still have issue with missing frei0r under plasma in 19.03. How do I test this snippet? Thanks!

**Comment by jtojnar** [comment]
You would need to add something similar to the kdenlive expression. Something like this:

```nix
  postInstall = ''
    wrapProgram $out/bin/kdenlive --prefix FREI0R_PATH : ${frei0r}/lib/frei0r-1
  '';
```

**Comment by chrissound** [comment]
Here is what I used to override the kdenlive package (thanks @jtojnar!).

```
   nixpkgs.config.packageOverrides = pkgs: {
     kdenlive = pkgs.kdenlive.overrideAttrs (oldAttrs: rec {
         postInstall = ''
          wrapProgram $out/bin/kdenlive --prefix FREI0R_PATH : ${pkgs.frei0r}/lib/frei0r-1
        '';
        nativeBuildInputs = oldAttrs.nativeBuildInputs or [] ++ [ pkgs.makeWrapper ];
     });
   };
```

I still have an issue though where I seem to be missing a bunch of rendering profiles - I have no 'lossy' profiles to render to. 

![2019-07-13-12:20:23](https://user-images.githubusercontent.com/1499652/61171568-08d77d80-a569-11e9-8ee6-b15d77c6878e.jpeg)

**Comment by jtojnar** [comment]
@chrissound Are there any more warnings about missing packages like above?

**Comment by chrissound** [comment]
Only relating to `ffplay` which I assume is not related. 
![2019-07-13-12:44:55](https://user-images.githubusercontent.com/1499652/61171809-18a49100-a56c-11e9-8f8e-ca7dca3b0a68.jpeg)

**Comment by raboof** [comment]
> Only relating to `ffplay` which I assume is not related

That warning goes away for me when I install `ffmpeg-full`

**Comment by chris-martin** [comment]
Sorry, I suppose I spoke too soon. After updating to the latest version of channel 20.03, things seem to be working fine for me now.

**Comment by jtojnar** [comment]
Closing then, as it should have been fixed by #83932 anyway.
